spec/integration/rails/finders_spec.rb

```diff
@@ -1095,6 +1099,71 @@ def show
             .to eq([other_table_hobby1.id, other_table_hobby2.id])
         end
       end
+
+      context 'when polymorphic many-to-many' do
+        let(:tag1) { Legacy::Tag.create!(name: 'One') }
+        let(:tag2) { Legacy::Tag.create!(name: 'Two') }
+
+        context 'when authors' do
+          before do
+            author1.tags << tag1
+            author2.tags << tag1
+            author2.tags << tag2
+          end
+
+          it 'still works' do
+            do_index({ include: 'tags' })
+            sl = d[0].sideload(:tags)
+            expect(sl.map(&:id)).to eq([tag1.id])
+            sl = d[1].sideload(:tags)
+            expect(sl.map(&:id)).to eq([tag1.id, tag2.id])
+          end
+        end
+
+        context 'when books' do
+          before do
+            book1.tags << tag1
+            book1.tags << tag2
+            book2.tags << tag1
+          end
+
+          it 'still works' do
+            allow(Legacy::BookResource).to receive(:validate_endpoints?) { false }
+            allow(controller).to receive(:resource) { Legacy::BookResource }
+            do_index({ include: 'tags' })
+            sl = d[0].sideload(:tags)
+            expect(sl.map(&:id)).to eq([tag1.id, tag2.id])
+            sl = d[1].sideload(:tags)
+            expect(sl.map(&:id)).to eq([tag1.id])
+          end
+        end
+
+        if ::Rails::VERSION::MAJOR >= 5
```

Comment:
?

Reply:
Under the hood this uses `.or`, which was introduced in Rails 5